swap_meet/vendor.py

```diff
@@ -1,2 +1,81 @@
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def swap_items(self, second_person, user_item, second_item):
+        if user_item in self.inventory and second_item in second_person.inventory:
+            # Add and remove from 'my' inventory
+            self.add(second_item)
+            self.remove(user_item)
+
+            # Add and remove from second vendor's inventory
+            second_person.add(user_item)
+            second_person.remove(second_item)
+            return True
+        return False
+
+    def swap_first_item(self, second_person):
+        if len(self.inventory) > 0 and len(second_person.inventory) > 0:
+            # return True/False by calling swap_items method
+            return self.swap_items(second_person, self.inventory[0], second_person.inventory[0])
+    
+    def get_by_category(self, category):
+        items = [item for item in self.inventory if item.category == category]
+        # Line 38 in integration tests wave 1/2/3 expects empty list if no category match
+        return items
+
+    def get_best_by_category(self, best_cat):
+        items_list = self.get_by_category(best_cat)
+        # max looks at item's self.condition value bc key points to lambda func
+        best_item = max(items_list, default=None, key=lambda x: x.condition)
```

Comment:
Fantastic use of `max()` with a lambda function! To improve readability, I might suggest shortening and specifying this line into something like:
```
best_item = max(items_list, key=lambda item: item.condition)
```